web/packages/teleport/src/Console/DocumentKubeExec/KubeExecDataDialog.tsx

```diff
@@ -0,0 +1,171 @@
+                <Flex gap={3}>
+                  <FieldInput
+                    value={namespace}
+                    placeholder="namespace"
+                    label="Namespace"
+                    autoFocus
+                    rule={requiredField('Namespace is required')}
+                    width="33%"
+                    onChange={e => setNamespace(e.target.value.trim())}
+                  />
+                  <FieldInput
+                    value={pod}
+                    placeholder="pod"
+                    label="Pod"
+                    rule={requiredField('Pod is required')}
+                    width="33%"
+                    onChange={e => setPod(e.target.value.trim())}
+                  />
+                  <FieldInput
+                    value={container}
+                    placeholder="container"
+                    label="Container (optional)"
+                    width="33%"
+                    onChange={e => setContainer(e.target.value.trim())}
+                  />
+                </Flex>
+                <Flex justifyContent="space-between" gap={3}>
+                  <FieldInput
+                    rule={requiredField('Command to execute is required')}
+                    value={execCommand}
+                    placeholder="/bin/bash"
+                    label="Command to execute"
+                    width="66%"
+                    onChange={e => setExecCommand(e.target.value)}
+                    toolTipContent={
+                      <Text>
+                        The command that will be executed inside the target pod.
+                      </Text>
+                    }
+                  />
+                  <Toggle
+                    isToggled={execInteractive}
+                    onToggle={() => {
+                      setExecInteractive(b => !b);
+                    }}
+                  >
+                    <Box ml={2} mr={1}>
+                      Interactive shell
+                    </Box>
+                    <ToolTipInfo>
+                      You can start an interactive shell and have a
+                      bidirectional communication with the target pod, or you
+                      can run one-off command and see its output.
+                    </ToolTipInfo>
+                  </Toggle>
+                </Flex>
+              </Box>
+            </DialogContent>
+            <DialogFooter>
+              <Flex justifyContent="space-between">
+                <ButtonSecondary
+                  type="button"
+                  width="45%"
+                  size="large"
+                  onClick={onClose}
+                >
+                  Close
+                </ButtonSecondary>
+                <ButtonPrimary
+                  type="submit"
+                  width="45%"
+                  size="large"
+                  onClick={e => {
+                    e.preventDefault();
+                    validator.validate() && podExec();
```

Comment:
I was like "How does this work without `<form>` having `onSubmit` on it?". I looked into the spec and it turns out that indeed, just having a `<button type="submit">` with an `onClick` is enough, TIL.

https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#implicit-submission